hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/WrappedWritableSingletonState.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.state;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * An implementation of {@link WritableSingletonState} that delegates to another {@link WritableSingletonState} as
+ * though it were the backend data source. Modifications to this {@link WrappedWritableSingletonState} are
+ * buffered, along with reads, allowing code to rollback by simply throwing away the wrapper.
+ *
+ * @param <T> the type of the state
+ */
+public class WrappedWritableSingletonState<T> extends WritableSingletonStateBase<T> {
```

Comment:
I think we will now also have to add `WrappedWritableQueueState`

Reply:
Created #6762 to track this.